### DISCUSSION TITLE
git problems when cloning

### DISCUSSION THREAD

**Discussion by MHz000** [rzeldent/esp32-smartdisplay#170] [Q&A]
On the entry page of Viaual Studio Code there is a button called "Clone Git Repository..." If I use it to recreate the demo project "https://github.com/rzeldent/esp32-smartdisplay-demo" then I get main.cpp in which all #incude <...> are shown as missing. The boards file is also empty.
I'm probably making a mistake when cloning the project because I don't understand it: "This library must reside in the <project>/boards directory so PlatformIo will automatically recognize these boards.
It is recommended to use git submodule to include these board definitions automatically" 
Please give me the correct git commands for the WIN10 environment.

**Comment by rzeldent** [comment]
Think this will help:
https://stackoverflow.com/questions/3796927/how-do-i-git-clone-a-repo-including-its-submodules

**Comment by MHz000** [comment]
rzeldent thanks for the tip. 
It's a bit easier if you go to the VScode home page, select view and then select Command Git: Clone (**Recursive**). Then paste the URL and specify the target folder for the clone. It will take a while and then I had to restart VScode.
I hope this helps others too
